src/main/java/org/jitsi/jibri/util/ProcessMonitor.kt

```diff
@@ -0,0 +1,33 @@
+package org.jitsi.jibri.util
+
+import org.jitsi.jibri.util.extensions.error
+import java.util.logging.Logger
+
+/**
+ * [ProcessMonitor] takes in a [MonitorableProcess] and a callback.  When
+ * its [run] method is executed it will check the status of the monitored
+ * process and, if it is no longer 'healthy' (the definition of 'healthy'
+ * is left to the [MonitorableProcess]'s implementation), [ProcessMonitor]
+ * will invoke the given [processUnhealthyCallback].
+ */
+class ProcessMonitor(
+        private val processToMonitor: MonitorableProcess,
+        private val processUnhealthyCallback: (exitCode: Int?) -> Unit
+    ) : Runnable {
+    private val logger = Logger.getLogger(this::class.qualifiedName)
+
+    /**
+     * Run a check of [processToMonitor] to check if it is alive.  If it is
+     * dead, [processUnhealthyCallback] will be invoked with the exit code from
+     * [processToMonitor] (or null, if there isn't one)
+     */
+    override fun run() {
+        try {
+            if (!processToMonitor.isHealthy()) {
+                processUnhealthyCallback(processToMonitor.getExitCode())
```

Comment:
I'm wondering why limit the callback to unhealthy only. Is it cleaner to limit this to unhealthy callbacks only at this point?

Reply:
yeah right now things are only interested in unhealthy-ness.